# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/WebGLErrorQueue.h
# Commit: ccfa767dba64
# Full Hash: ccfa767dba644dc193e0246eb8e8ff3377e8b8a5
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2020-01-09 03:40:18
# Regressor Bug: 1477756
# File Overlap Count: 1
# Description:
#   Bug 1477756 - Client-side bindings mirror for precise CC, and merge similar codepaths. r=handyman
#   
#   * Context loss using RAII
#   * Move Program reflection Client-side
#   
# ==============================================================================

diff -r 71c122ac0ca7 -r ccfa767dba64 dom/canvas/WebGLErrorQueue.h
--- a/dom/canvas/WebGLErrorQueue.h	Wed Jan 08 22:19:14 2020 +0000
+++ /dev/null	Thu Jan 01 00:00:00 1970 +0000
@@ -1,58 +0,0 @@
-/* -*- Mode: C++; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 2 -*- */
-/* This Source Code Form is subject to the terms of the Mozilla Public
- * License, v. 2.0. If a copy of the MPL was not distributed with this
- * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
-
-#ifndef WEBGLERRORQUEUE_H_
-#define WEBGLERRORQUEUE_H_
-
-#include "WebGLCommandQueue.h"
-
-namespace mozilla {
-
-class ClientWebGLContext;
-
-using mozilla::webgl::PcqStatus;
-
-enum WebGLErrorCommand {
-  CreationError,
-  OnLostContext,
-  OnRestoredContext,
-  Warning
-};
-
-/**
- * This is the source for the WebGL error and warning queue.  When an
- * asynchronous error or warning occurs in the host, such as the GL context
- * being lost or a diagnostic message being printed, it is inserted into this
- * queue to be asynchronously retrieved by the client in the content process,
- * where it is communicated to the relevant DOM objects.
- */
-using HostWebGLErrorSource = CommandSource<WebGLErrorCommand>;
-
-/**
- * This is the sink for the WebGL error and warning queue.
- * Before this sink can handle events, it needs to have its ClientWebGLContext
- * set.
- */
-class ClientWebGLErrorSink : public CommandSink<WebGLErrorCommand> {
- public:
-  ClientWebGLErrorSink(UniquePtr<Consumer>&& aConsumer);
-
-  void SetClientWebGLContext(RefPtr<ClientWebGLContext>& aClientContext);
-
- protected:
-  bool DispatchCommand(WebGLErrorCommand command) override;
-
-  RefPtr<ClientWebGLContext> mClientContext;
-};
-
-/**
- * Factory for the WebGL error and warning queue.
- */
-using WebGLErrorQueue =
-    CommandQueue<HostWebGLErrorSource, ClientWebGLErrorSink>;
-
-}  // namespace mozilla
-
-#endif  // WEBGLERRORQUEUE_H_